official/resnet/resnet_run_loop.py

```diff
@@ -287,34 +309,37 @@ def exclude_batch_norm(name):
       eval_metric_ops=metrics)
 
 
-def validate_batch_size_for_multi_gpu(batch_size):
+def compute_per_device_batch_size(batch_size, use_distribution_strategy,
+                                  gpus_for_distribution_strategy):
   """For multi-gpu, batch-size must be a multiple of the number of GPUs.
 
-  Note that this should eventually be handled by replicate_model_fn
+  Note that this should eventually be handled by DistributionStrategies
   directly. Multi-GPU support is currently experimental, however,
   so doing the work here until that feature is in place.
 
   Args:
-    batch_size: the number of examples processed in each training batch.
+    batch_size: Global batch size to be divided among devices.
+    use_distribution_strategy: Whether DistributionStrategies API is used.
+    gpus_for_distribution_strategy: How many GPUs are used with
+      DistributionStrategies.
+
+  Returns:
+    Batch size per device.
 
   Raises:
-    ValueError: if no GPUs are found, or selected batch_size is invalid.
+    ValueError: if batch_size is not divisible by number of devices
   """
-  from tensorflow.python.client import device_lib  # pylint: disable=g-import-not-at-top
-
-  local_device_protos = device_lib.list_local_devices()
-  num_gpus = sum([1 for d in local_device_protos if d.device_type == 'GPU'])
-  if not num_gpus:
-    raise ValueError('Multi-GPU mode was specified, but no GPUs '
-                     'were found. To use CPU, run without --multi_gpu.')
-
-  remainder = batch_size % num_gpus
-  if remainder:
-    err = ('When running with multiple GPUs, batch size '
-           'must be a multiple of the number of available GPUs. '
-           'Found {} GPUs with a batch size of {}; try --batch_size={} instead.'
-          ).format(num_gpus, batch_size, batch_size - remainder)
-    raise ValueError(err)
+  if use_distribution_strategy and gpus_for_distribution_strategy > 1:
```

Comment:
As above, I don't see the value of the dual check here. If num_gpus > 1 should be sufficient, since we don't allow dist strat with num_gpu <= 1, and we don't have another way of doing multi-GPU.

Reply:
+1